src/Http/WebUtilities/src/MultipartSectionPipeReader.cs

```diff
@@ -73,11 +73,21 @@ public override void AdvanceTo(SequencePosition consumed, SequencePosition exami
                 throw new InvalidOperationException("AdvanceTo can only be called once per read operation.");
             }
 
-            if (!_boundaryFound && examined.Equals(_bodyBuffer.End))
+            if (examined.Equals(_bodyBuffer.End))
             {
-                // The caller has seen all of the available data. We may have a partial boundary match in the
-                // unconsumed buffer and we need more data to continue.
-                _pipeReader.AdvanceTo(consumed, _unconsumedData.End);
+                // The caller has seen all of the available data.
+                if (!_boundaryFound)
+                {
+                    // We may have a partial boundary match in the
+                    // unconsumed buffer and we need more data to continue.
+                    _pipeReader.AdvanceTo(consumed, _unconsumedData.End);
+                }
+                else
+                {
+                    // We reached the end of the section, we should advance to the end.
+                    _pipeReader.AdvanceTo(_unconsumedData.End);
+                    _finalLineConsumed = true;
```

Comment:
This solves an issue with _finalLineConsumed is false (and probably shouldn't be), I am not sure if it's the correct solution though, and I might be missing an edge case.